From 041605bc50c7f06cb4aab5c3fd7c1189854c810c Mon Sep 17 00:00:00 2001
From: Philip Chimento <philip.chimento@gmail.com>
Date: Tue, 27 Aug 2013 23:46:22 -0700
Subject: [PATCH 3/5] [quartz] NULL check on default keymap

Ensure that Quartz's default keymap is not accessed before it is
created.
---
 gdk/quartz/gdkkeys-quartz.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gdk/quartz/gdkkeys-quartz.c b/gdk/quartz/gdkkeys-quartz.c
index 8ecf79e..d2b5110 100644
--- a/gdk/quartz/gdkkeys-quartz.c
+++ b/gdk/quartz/gdkkeys-quartz.c
@@ -492,7 +492,8 @@ update_keymap (void)
         p[0] = known_numeric_keys[i].keypad_keyval;
     }
 
-  g_signal_emit_by_name (default_keymap, "keys-changed");
+  if (default_keymap != NULL)
+    g_signal_emit_by_name (default_keymap, "keys-changed");
 }
 
 static PangoDirection
-- 
1.7.12.4 (Apple Git-37)

